### PR TITLE
Add inclination method for QgsPointV2

### PR DESCRIPTION
**Pull request by lbartoletti**
The equivalent of the azimuth calculation for vertical angles (Z / 3D).
A method that will be widely used for geometries.

### CODE REVIEW

File: src/core/qgsexpression.cpp [hunk 1]
```diff
@@ -2921,6 +2921,25 @@ static QVariant fcnProject( const QVariantList &values, const QgsExpressionConte
   return QVariant::fromValue( QgsGeometry( new QgsPointV2( newPoint ) ) );
 }
 
+static QVariant fcnInclination( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  QgsGeometry fGeom1 = getGeometry( values.at( 0 ), parent );
+  QgsGeometry fGeom2 = getGeometry( values.at( 1 ), parent );
+
+  const QgsPointV2 *pt1 = static_cast<const QgsPointV2 *>( fGeom1.geometry() );
```
Comment (nyalldawson) [inline src/core/qgsexpression.cpp]: better make these dynamic_casts, or move the wkb type check before this static_cast

**Comment by nyalldawson** [comment]
Looking good - just a tiny change for safety

File: src/core/qgsexpression.cpp (hunk 1 above)
Comment (m-kuhn) [inline src/core/qgsexpression.cpp]: We should probably make a `geometry_cast` that does the type check.

Comment (lbartoletti) [inline src/core/qgsexpression.cpp]: I have changed with a dynamic_cast.

**Pull request merged**